server/stream.go

```diff
@@ -2880,21 +2887,26 @@ func (mset *stream) processJetStreamMsg(subject, reply string, hdr, msg []byte,
 		// Expected last sequence per subject.
 		if seq, exists := getExpectedLastSeqPerSubject(hdr); exists {
 			// TODO(dlc) - We could make a new store func that does this all in one.
-			_, lseq, _, _, _, err := mset.store.LoadLastMsg(subject)
+			var smv StoreMsg
+			var fseq uint64
+			sm, err := mset.store.LoadLastMsg(subject, &smv)
+			if sm != nil {
+				fseq = sm.seq
```

Comment:
So clearly here, it looks like you have LoadLastMsg() that can return an error and a non nil sm..

Reply:
No, it is doing the correct thing, would you prefer this to read err == nil?

Reply:
The problem is that after this check you check for errors but in the form of `if err != nil || fseq != seq` so I guess you have a situation where you check for both error or sequence mismatch. So I guess it's fine like that..